Gooliajulia/native signup

gooliajulia opened the pull request:
Refactors Sign Up screen to React Native

Notes:
- Will need to replace certain sections of code with custom components once components are refactored to native: Modal, SingleActionButton, DoubleActionButton, Form (tbd)
- checkEmail is not case sensitive. For example 'firstuser@mail.com' will pass but 'firstUser@mail.com' will not (neither should pass). Should we convert all user emails to lowercase before storing them on backend?

gooliajulia committed:
structure, layout and lofi styling

gooliajulia committed:
add navigation functionality to Sign In link

gooliajulia committed:
first round refactor

gooliajulia committed:
update successful login modal and rerouting

gooliajulia committed:
remove non native commented out code

gooliajulia committed:
Delete Podfile.lock

gooliajulia committed:
remove modified podfile from pull request

gooliajulia committed:
remove modified xcode fiel from pull request

gooliajulia committed:
merge

gooliajulia committed:
restore podfile

gooliajulia committed:
remove modified pod from PR

gooliajulia committed:
remove modified package.json from PR

gooliajulia committed:
remove modified package-lock.json from PR

gooliajulia committed:
basic structure, functionality and styling refactored to native

gooliajulia committed:
remove commented out non native code

yeezick commented on native/screens/SignUp.jsx:
```diff
@@ -1,180 +1,309 @@
-//     password: "",
-//   });
+export const SignUp = ({ navigation }) => {
+  // redux
```
you don't need comments when your code can speak for you

yeezick replied:
also, i know this doesn't exactly apply but something i heard recently that i enjoyed:
long, descriptive names >>>> long, descriptive comments

yeezick commented on native/screens/SignUp.jsx:
```diff
@@ -1,180 +1,309 @@
+    setSuccessModalVisible(false);
+    // need to test this params funcionality on edit profile screen
+    navigation.navigate(screen, {
+      id: id,
```
although there is no other ID on this file, it may not be obvious to the dev that accesses this `id` param on the next screen, so let's be semantic about passing in the userID to the next screen.

yeezick commented on native/screens/SignUp.jsx:
```diff
@@ -1,180 +1,309 @@
+    lastNameInputRef, 
+    emailInputRef, 
+    passwordInputRef, 
+    confirmPasswordInputRef } = createRef();
```
i have not touched on refs at all yet, i'd love to schedule a 1:1 with you whenever you have time to teach me :)

yeezick commented on native/screens/SignUp.jsx:
```diff
@@ -1,180 +1,309 @@
+        </View>
+      </Modal>
+      <Text style={styles.title} >Create an Account</Text>
+      {/* Replace with Form componenet? */}
```
i would say no because the textInputs below are complex and the Form component is already very hard to work with; we learned the hard way that custom form components are harder than learning to program lol

yeezick commented on native/screens/SignUp.jsx:
```diff
@@ -1,180 +1,309 @@
+        <TextInput 
+          value={newUser.first_name}
+          style={styles.input}
+          onChangeText={(firstName) => setNewUser((prevState) => {
```
take a look at the handler in `./services/utils/handlers` to see if that function can work for you. otherwise, create a generic function in this component to handle the onChangeText and DRY up the logic

yeezick commented on native/screens/SignUp.jsx:
```diff
@@ -1,180 +1,309 @@
+          autoCapitalize="none"
+          ref={emailInputRef}
+          onFocus={()=>setEmailError(null)}
+          onBlur={() => handleEmailCheck()}
```
applauding your superb uses of `onFocus` / `onBlur`

yeezick commented on native/screens/SignUp.jsx:
```diff
@@ -1,180 +1,309 @@
+
+const styles = StyleSheet.create({
+  modalContainer: {
+    height: '100%',
```
please try to alphabetize your properties & classes; it will pay off in the long run in case we ever have to come back to update the styling

yeezick commented:
And i thought `roulette` was hairy! Very nice work @gooliajulia. Before I merge this, I'm going to merge my PR so that you have the `SingleActionButton` as well as the Redux Store all set up. This way you can use the user in the Redux store instead of using `verify` to grab the user id. (if i didnt understand your intention with the verify method, feel free to let me know)

yeezick committed:
pulling down changes from current native0development to resolve conflicts and use latest code
Merge branch 'native-development' of https://github.com/yeezick/Bootcamper into gooliajulia/native-signin

yeezick committed:
- update root useEffect to set up redux store on app render based on user verification
- remove comments from SignIn
- configure sign in to make the api call itself instead of thru redux-async-dispath method due to poor error handling and lack of tangible response from that method, it is now deprecated
- update signIn api function call to handle unsuccessful login attempts better
- replace onChangeText function with pre-built handler
- added error handling for empty payload in redux updateUser

yeezick committed:
alphabetize stylsheet

yeezick committed:
Merge pull request #57 from yeezick/gooliajulia/native-signin

Gooliajulia/native signin

yeezick committed:
- add email format validation to user schema
- include a few more edge cases for client-side error handling
- use the base signUp api function vs redux-built function due to state lifecycle errors
- add email validation to signUp
- replace onChangeText with changeText handler from util

yeezick committed:
alphabetize stylesheet

yeezick commented:
@gooliajulia reviewed and tested with minor updates to comply with what i've learned from reviewing your SignUp PR

I didn't have the chance to convert your touchable opacity to the singleActionButton but it will be worth to revise SingleActionButton to see how to make it work best across the app.
I'd also like to see if we can somehow simplify the file by breaking it into smaller components for legibility; UserProfile has a small example of this 
Good work regardless!
